Terminal.Gui/Application/ApplicationNavigation.cs

```diff
@@ -55,14 +55,14 @@ internal void SetFocused (View? value)
     /// <param name="start"></param>
     /// <param name="view"></param>
     /// <returns></returns>
-    public static bool IsInHierarchy (View start, View? view)
+    public static bool IsInHierarchy (View? start, View? view)
     {
         if (view is null)
         {
             return false;
         }
 
-        if (view == start)
+        if (view == start || start is null)
```

Comment:
How about replacing all of these conditions with a switch similar to the following (or, if all cases are just returns, then the equivalent switch expression, which is super-compact):

```csharp
switch(view,start)
  case (null,_):
    return false;
  case (_,null):
  case (_,_) when view == start:
    return true;
//etc
}
```

Reply:
I'm still getting used to parsing switch expressions... at a glance they are un-readable to me. See the 1st tenet below:

## Tenets for [gui-cs](www.github.com/gui-cs) Code Style (Unless you have better ones)

* **Six-Year-Old Reading Level** - Our code style is biased towards code readability and away from terseness. This is *Systems Software* and needs to stand the test of time. Code should be structured and use variable names that make it readable by a 6-year-old, and comments in code are encouraged. 
* **Consistency, Consistency, Consistency** - We adopt and document our standards for code style and then enforce them ruthlessly. For example, we require code reviews to pay attention to code style, not just functionality. 
* **Don't be Weird** - Like all developers we have opinions, but our opinions on code style are tempered by existing standards. We are biased towards code style that used by Microsoft and other leading dotnet developers. For example, we choose 4 spaces for indentation instead of 8.
* **Set and Forget** - We embrace and encourage the use of technology that makes it easy for contributors to apply best-practice code-style, such as ReSharper. As we do so we are mindful that tools can cause hidden issues and merge hell.
* **Documentation is the Spec** - We care deeply about providing delightful developer documentation and are sticklers for grammar and clarity. If the code and the docs conflict, we are biased to believe what we wrote in the API documentation. This drives a virtuous cycle of clear thinking.

from: https://github.com/gui-cs/Terminal.Gui/pull/3207#issuecomment-1908203789

(Which I meant to pull out into `CONTRIBUTING.md` but forgot. Will do).

Reply:
All good either way, as far as I'm concerned.

I can certainly empathize with learning unfamiliar constructs or things one might just not be highly confident with at a given time. Personally, having worked with C# over its whole existence has kept me on my toes because it never stops changing and I learn something new darn near every day about either the language or .net, still, almost 24 years later! 🤯 

OOOO thanks for moving that whenever you get to it. I always thought it was in a sub-optimal and slightly not-so-discoverable place, before. 👍 

#### Philosophical trivialities, ramblings, and opinionated blathering

Welcome to the wasting of the next...hmm...Maybe 1-5 minutes of your life? 😅 

As to the applicability of the referenced tenet _here_, specifically, ehhhh... I don't think it applies, when we're talking about standard, idiomatic C#, especially with constructs that are not remotely new to the language. C# is pretty big, .net is bigger, and programming is ridicu-hard when you step back and look at what we do. One could pretty arbitrarily pick almost any programming concept and call it complex while someone else thinks it's the simplest and best thing since sliced bread (which I, incidentally, don't find to be so awesome as to deserve its own idiomatic expression 😆).

Things like the concept of readability are more subjective than objective, in particular. I, personally, tend to find **well-written** and **appropriately-used** switches more readable, less prone to easily-made and difficult-to-locate mistakes, clearer in how many cases there actually are,[^myGodItsFullOfStars] and easier to parse in general since they are fully declarative, by specification of the language. I especially like them when they reduce otherwise large and widely-separated conditionals or, on the other end of the spectrum, simple but very subtly different conditions in interdependent statements down to a literal truth table. But those emphasized terms are critical components of it, as you can just as easily make a monstrosity that is hella indecipherable to any of us ugly bags of mostly water.

There are times when one of the three (if/else if/else, switch statement, or switch expression) may have technical advantages as well, but if that might matter I'll point that fact out - definitely will if it's important enough.

Which is why _this one_ is just written as a soft suggestion rather than an imperative with technical justification (you know - in my signature ultra-terse, concise, and devoid of dad joke-esque humor way).

As a reminder: Usually, when I make a suggestion like this, it's as much for presenting alternatives with potential benefits (I won't make any that are _completely_ pointless) as it is for enrichment of those reading the review comment, be they the author of the code, other project members, other people who randomly stumble upon it due to a google search that turns something up that makes them have their lightbulb moment, or myself 6 months from now when I can't figure something out due to coming down with a relapse of stoopid. 😅🤷‍♂️ 

[^myGodItsFullOfStars]: `if` often hides soooooo many possibilities unless it truly is a binary choice and the `else` is fully valid no matter which individual part of the boolean condition made it evaluate to false, and the complexity of that grows polynomially per condition and quadratically component of each condition.